src/components/subgoal/Subgoal.tsx

```diff
@@ -2,25 +2,17 @@ import { trpc } from "@/client/lib/trpc";
 import { Subgoal } from "@/types/global";
 import Box from "@mui/material/Box";
 import Button from "@mui/material/Button";
+import { useState } from "react";
+import { SubgoalAssignmentModal } from "./Subgoal-Assignment-Modal";
 
 interface SubgoalProps {
   subgoal: Subgoal;
 }
 
 const Subgoals = ({ subgoal }: SubgoalProps) => {
+  const [isAssignmentModalOpen, setIsAssignmentModalOpen] = useState(false);
   const task = trpc.iep.tempAddTaskToSelf.useMutation();
 
-  // TODO: add form to assign to my paras
-  // const assignToPara = (event: React.FormEvent<HTMLFormElement>) => {
-  //   event.preventDefault();
-  //   const data = new FormData(event.currentTarget);
-
-  //   task.mutate({
-  //     subgoal_id: subgoal.subgoal_id,
-  //     assignee_id: data.get("assignee_id") as string,
-  //     due_date: new Date(data.get("due_date")) as Date
```

Comment:
Let's be sure to test this when we implement it, to ensure it comes through as the correct date.